### PR TITLE
[Feature] Web Support

### PR DESCRIPTION
**Pull request by stevensJourney**
# Overview

This adds Web support to the stable `sqlite_async` package. 

The use of Web support inside the package is still in a beta release. 

Native operations should not be affected by these changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,4 +74,29 @@ void main() async {
...
+# Web
+
+Web support is provided by the [Drift](https://github.com/powersync-ja/drift/pull/1) library. Detailed instructions for compatibility and setup are listed in the link.
+
+Web support requires Sqlite3 WASM and Drift worker Javascript files to be accessible via configurable URIs.
+
+Default URIs are shown in the example below. URIs only need to be specified if they differ from default values.
+
+Watched queries and table change notifications are only supported when using a custom Drift worker which is compiled by linking
+https://github.com/powersync-ja/drift/pull/1.
```
Comment (rkistner) [inline README.md:88]: This looks outdated since we're not depending on Drift anymore

Comment (stevensJourney) [reply README.md]: This has been updated.

File: lib/src/web/web_sqlite_open_factory.dart [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    throw UnimplementedError(
+        'Direct access to CommonDatabase is not available on web.');
+  }
+
+  @override
+
+  /// Currently this only uses the Drift WASM implementation.
+  /// The Drift SQLite package provides built in async Web worker functionality
+  /// and automatic persistence storage selection.
+  /// Due to being asynchronous, the under laying CommonDatabase is not accessible
```
Comment (rkistner) [inline lib/src/web/web_sqlite_open_factory.dart:48]: This also doesn't seem accurate anymore.

**Review by rkistner — APPROVED** [review]

File: lib/src/web/web_sqlite_open_factory.dart (hunk 2 above)
Comment (stevensJourney) [reply lib/src/web/web_sqlite_open_factory.dart]: Thanks! I've searched for and updated all mentions of Drift now.

**Review by rkistner — APPROVED** [review]

**Review by benitav — APPROVED** [review]

**Pull request merged**